apis/nucleus/src/index.js

```diff
@@ -344,6 +344,44 @@ function nuked(configuration = {}) {
          * @typedef {function(function)} ReceiverFunction A callback function which receives another function as input.
          */
 
+        /**
+         * @typedef {object} c SwitchButtonConfig
```

Comment:
For now `@interface` is needed to be used for get the typescript definitions generated properly in the Nebula.js repo.

```suggestion
         * @interface SwitchButtonConfig
         * @property {string} option The option to be toggled. Currently supporting: 'checkboxes', 'search', 'toolbar', 'rangeSelect', 'dense'.
         * @property {boolean=} [invert=false] Invert the switch button's value when applied on the option.
         * @property {boolean=} [startOn=false] Start the switch turned on (note that this will not change the initial value of the option).
         * @property {string} [label] A label for the switch.
         * @property {string} [helperText] A helper text label for the switch.
         * @property {object} [iconOn] CSS styling for the switch's on state (applied on span::after).
         * @property {object} [iconOff] CSS styling for the switch's off state (applied on span::before).
         * @property {function} [onChange] A function for defining a custom callback or for setting multiple properties using the provided setOptions function.
```

Reply:
Ok, yes I noticed the discussion around this 👍